Lib/SmartPtr.hpp

```diff
@@ -63,6 +63,7 @@ class SmartPtr {
   SmartPtr(const SmartPtr& ptr) : _obj(ptr._obj), _refCnt(ptr._refCnt)
   {
     if(_obj && _refCnt) {
```

Comment:
"should" always be true - otherwise the memory has already been deleted. Also a classic thread-safety bug (need an atomic refcount), might canary later!